psydac/linalg/kernels.py

```diff
@@ -443,7 +443,7 @@ def axpy_3d(alpha: 'T', x: "Tarray", y: "Tarray",shape0:'int64',shape1:'int64',s
     for i1 in range(0, shape0, 1):
         for i2 in range(0, shape1, 1):
             for i3 in range(0, shape2, 1):
```

Comment:
You can just write
```suggestion
    for i1 in range(shape0):
        for i2 in range(shape1):
            for i3 in range(shape2):
```
But why do we need to pass  `shape0`, `shape1` and `shape2` to this function? One can just write
```python
    n1, n2, n3 = x.shape

    for i1 in range(n1):
        for i2 in range(n2):
            for i3 in range(n3):
```

Reply:
This commit correct it 81d63c011b84ef5f43e5a8e9acf5e1b6b133c22a